functional/churn/README.md

```diff
@@ -0,0 +1,36 @@
+# churn
+Test targeting garbage collector(s).
+The suite is able to operate with all GCs in OpenJDK. If your implementation is missing some, the calls to them will fall. If you have some additional one, an enabling wrapper must be created first.
+
+## Running churn locally
+clone https://github.com/rh-openjdk/churn read  https://github.com/rh-openjdk/churn/blob/master/README and run run.sh
+
+### Setting up the environment:
+Java required. Maven recommended, churn can be compiled via direct javac if needed
+Java is used from JAVA_HOME or guessed from PATH.
+run.sh can use OTOOL_garbageCollector and OTOOL_JDK_VERSION instead of GC argument. See https://github.com/rh-openjdk/churn/blob/master/README for details.
+See https://github.com/rh-openjdk/churn/tree/master/bin for list of supported GCs. The compressed ops in upstream run can be controlled by runner or NOCOMP=-nocoops variable (for run.sh)
+
+## Running via aqavit
+
+The support for compressed ops is handled by aqavit itself. It is currently not sure if churn will be able to honor it. If not, churn will be fixed.
+Similarly the java version and JAVA_HOME are handled by aqavit.
+
+### Executing the testsuite
+The `BUILD_LIST of functional/churn` contains three targets:
+ * _churn_1m_allGCs
+ * _churn_5h_allGCs
+ * _churn_custom
+
+There are major differences in them:
+ * churn_1m_allGCs - is testing ground, which runs each GC only for aprox 10 seconds, to simply see if the setup works
+ * churn_5h_allGCs - Is running each GC a bit over, which runs each GC for aprox hour and half. A minimum, which can find some real GC issue.
+ * churn_custom - this one needs you to set and export CHURN_GCS and CHURN_DURATION to select GC(or GCs) and duration. So it allows you to test your custom GC - if churn supports that, despite of anybody else (eg `default` or `ALL` thinks)
```

Comment:
```suggestion
 * churn_custom - this one is to support development, when run on the commandline you need to export CHURN_GCS and CHURN_DURATION to select GC(or GCs) and duration. So it allows you to test your custom GC - if churn supports that, despite what other churn options suggest (eg `default` or `ALL` thinks).  When running in a Jenkins Grinder job, can use the APPLICATION_OPTIONS parameter to set CHURN_GCS value and CHURN_DURATION defaults to 5 minutes.
```

Reply:
I had adapt this only partially, as I had not (yet!) followed the APPLICATION_OPTIONS and 5 minutes as you requested (subject of another discussion)